File: RockPaperScissor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Game
{
    public class RockPaperScissor
    {
        private static Dictionary<int, string> Choices = new Dictionary<int, string> { { 1, "Rock" }, { 2, "Paper" }, { 3, "Scissor" } };

        public static void Play()
        {
            int compScore = 0, userScore = 0, winningScore = 3;

            int userChoice;

            Console.WriteLine("Game Started".ToUpper());

            while (compScore < winningScore && userScore < winningScore)
            {
                Console.WriteLine("Enter Your Choice");
                Console.WriteLine("1:Rock");
                Console.WriteLine("2:Paper");
                Console.WriteLine("3:Scissor");

                userChoice = Int32.Parse(Console.ReadLine());

                if (userChoice != 1 && userChoice != 2 && userChoice != 3)
                {
                    Console.WriteLine("You made the wrong Choice!");
                    continue;
                }

                int randomNumber = new Random().Next(1, 3);

                int result = CheckWinner(userChoice, randomNumber);

                if (result == 0)
                {
                    Console.WriteLine("Draw!".ToUpper());
                }
                else if (result > 0)
                {
                    userScore++;
                    Console.WriteLine("You won!".ToUpper());
                }
                else
                {
                    compScore++;
                    Console.WriteLine("You Lost!".ToUpper());
                }
            }


            if (userScore > compScore)
            {
                Console.WriteLine("You Won The Game!");
            }
            else
            {
                Console.WriteLine("You Lost the Game!");
            }
        }

        private static int CheckWinner(int userChoice, int compChoice)
        {
            Console.WriteLine($"Computer Chose {Choices[compChoice]}");

            if (userChoice == compChoice)
            {
                return 0;
            }

            switch (userChoice)
            {
                case 1:
                    return compChoice == 2 ? -1 : 1;
                case 2:
                    return compChoice == 3 ? -1 : 1;
                case 3:
                    return compChoice == 1 ? -1 : 1;
                default:
                    throw new ArgumentException();
            }
        }

    }
}

File: Golang/rockPaperScissor/rockPaperScissor.go
package rockPaperScissor

import (
	"bufio"
	"errors"
	"fmt"
	"math/rand"
	"os"
	"strconv"
)

type RockPaperScissor struct{}

func (*RockPaperScissor) Play() {

	stdScanner := bufio.NewScanner(os.Stdin)

	var compScore, userScore, winningScore int = 0, 0, 3

	var userChoice int

	fmt.Println("GAME STARTED")

	for compScore < winningScore && userScore < winningScore {
		fmt.Println("Enter Your Choice")
		fmt.Println("1.Rock")
		fmt.Println("2.Paper")
		fmt.Println("3.Scissor")

		stdScanner.Scan()

		userChoice, _ = strconv.Atoi(stdScanner.Text())

		if userChoice != 1 && userChoice != 2 && userChoice != 3 {
			fmt.Println("You made the wrong choice!")
			continue
		}

		randomNumber := rand.Intn(2) + 1

		result, err := checkWinner(userChoice, randomNumber)

		if err == nil {
			if result == 0 {
				fmt.Println("Draw!")
			} else if result == 1 {
				userScore++
				fmt.Println("You Won!")
			} else if result == 2 {
				compScore++
				fmt.Println("You Lost!")
			}
		} else {
			fmt.Println(err.Error())
		}
	}

	if userScore > compScore {
		fmt.Println("You Won the Game!")
	} else {
		fmt.Println("You Lost the Game!")
	}
}

func checkWinner(userChoice int, compChoice int) (int, error) {

	choices := map[int]string{1: "Rock", 2: "Paper", 3: "Scissor"}

	fmt.Println("Computer chose " + choices[compChoice])

	if userChoice == compChoice {
		return 0, nil
	}

	switch userChoice {
	case 1:
		if compChoice == 2 {
			return 1, nil
		} else {
			return -1, nil
		}
	case 2:
		if compChoice == 3 {
			return -1, nil
		} else {
			return 1, nil
		}
	case 3:
		if compChoice == 1 {
			return -1, nil
		} else {
			return 1, nil
		}
	}

	return -2, errors.New("wrong choice")
}
